### ISSUE TITLE
TNRS result for 'Primates' has redundant match

### ISSUE THREAD

**Issue by jar398** [OpenTreeOfLife/taxomachine#73]
The matches list for Primates has two elements, but they are the same. The problem is that applications will assume that there's an ambiguity when there isn't.

```
    {u'context': u'Mammals',
     u'governing_code': u'ICZN',
     u'includes_approximate_matches': False,
     u'includes_deprecated_taxa': False,
     u'includes_dubious_names': False,
     u'matched_name_ids': [u'Mammalia', u'Rodentia', u'Primates'],
     u'results': [{u'id': u'Mammalia',
                   u'matches': [{u'flags': [u'EDITED'],
                                 u'is_approximate_match': False,
                                 u'is_deprecated': False,
                                 u'is_dubious': False,
                                 u'is_synonym': False,
                                 u'matched_name': u'Mammalia',
                                 u'matched_node_id': 3538050,
                                 u'nomenclature_code': u'ICZN',
                                 u'ot:ottId': 244265,
                                 u'ot:ottTaxonName': u'Mammalia',
                                 u'rank': u'',
                                 u'score': 1.0,
                                 u'search_string': u'mammalia',
                                 u'synonyms': [u'mammals',
                                               u'Xenostylus',
                                               u'Mammalia',
                                               u'Megahippus'],
                                 u'unique_name': u'Mammalia'}]},
                  {u'id': u'Rodentia',
                   u'matches': [{u'flags': [u'EDITED'],
                                 u'is_approximate_match': False,
                                 u'is_deprecated': False,
                                 u'is_dubious': False,
                                 u'is_synonym': False,
                                 u'matched_name': u'Rodentia',
                                 u'matched_node_id': 3540790,
                                 u'nomenclature_code': u'ICZN',
                                 u'ot:ottId': 864593,
                                 u'ot:ottTaxonName': u'Rodentia',
                                 u'rank': u'',
                                 u'score': 1.0,
                                 u'search_string': u'rodentia',
                                 u'synonyms': [u'Rodentia', u'rodents'],
                                 u'unique_name': u'Rodentia'}]},
                  {u'id': u'Primates',
                   u'matches': [{u'flags': [u'EDITED'],
                                 u'is_approximate_match': False,
                                 u'is_deprecated': False,
                                 u'is_dubious': False,
                                 u'is_synonym': False,
                                 u'matched_name': u'Primates',
                                 u'matched_node_id': 3551712,
                                 u'nomenclature_code': u'ICZN',
                                 u'ot:ottId': 913935,
                                 u'ot:ottTaxonName': u'Primates',
                                 u'rank': u'',
                                 u'score': 1.0,
                                 u'search_string': u'primates',
                                 u'synonyms': [u'primate',
                                               u'Primates',
                                               u'Primata',
                                               u'primates'],
                                 u'unique_name': u'Primates'},
                                {u'flags': [u'EDITED'],
                                 u'is_approximate_match': False,
                                 u'is_deprecated': False,
                                 u'is_dubious': False,
                                 u'is_synonym': True,
                                 u'matched_name': u'primates',
                                 u'matched_node_id': 3551712,
                                 u'nomenclature_code': u'ICZN',
                                 u'ot:ottId': 913935,
                                 u'ot:ottTaxonName': u'Primates',
                                 u'rank': u'',
                                 u'score': 1.0,
                                 u'search_string': u'primates',
                                 u'synonyms': [u'primate',
                                               u'Primates',
                                               u'Primata',
                                               u'primates'],
                                 u'unique_name': u'Primates'}]}],
     u'taxonomy': {u'author': u'open tree of life project',
                   u'source': u'ott2.8',
                   u'weburl': u'https://github.com/OpenTreeOfLife/opentree/wiki/Open-Tree-Taxonomy'},
     u'unambiguous_name_ids': [u'Mammalia', u'Rodentia', u'Primates'],
     u'unmatched_name_ids': []}
```

**Comment by jimallman** [comment]
@jar398 It seems this was caused by two instances of 'primates' in the synonym list:
`[u'primate', u'Primates', u'Primata', u'primates']`
These differ only by case, which is presumably ignored by our search.

N.B. that one of these no longer appears on **devapi**, and only one match is returned by our TNRS. The response now shows this synonym list: `[ "Primata", "primate", "Primates" ]`.

I'd say we should close this, but it would be helpful to sweep for other cases where there are identical synonyms for a taxon (or differing only by case or whitespace). See for example `Orthotomus cucullatus`, with this synonym list: `[ "Orthotomus cuculatus", "Orthotomus cuculatus", "Phyllergates cucullatus", "Phyllergates cuculatus", "Orthotomus cucullatus" ]`

Note that the first two synonyms are identical, which shows two matching results from TNRS.

**Comment by jar398** [comment]
Will fix.

**Comment by jar398** [comment]
To be more precise: there is a new situation (*) where a taxon A can have
two synonym records both for name A (different synonym type in the two
sources). I am happy to fix that. But the other case (**), with similar but
not identical-named synonyms is not something that should be fixed by
tweaking the synonyms file. And the problem would have had to be present in
OTT 2.9 - making me wonder (1) why is this issue coming up now and didn't
earlier, (2) maybe this should be fixed in the UI, which really ought to
offer a selection of taxa (OTT ids), with each taxon listed once regardless
of how many similar synonyms it has. In fact I don't understand how the UI
could do anything other than offer a choice among taxa. Maybe I'm not
getting how this works.

In other words, (**) should not be fixed in the taxonomy, and nobody wants
to makes changes to the TNRS, so that means it would have to be fixed in
the front end. I expect a fix for (**) would also be a fix for (*).

**Comment by jimallman** [comment]
> But the other case (**), with similar but not identical-named synonyms...

To be clear, I've assumed that 'Primates' and 'primates' are identical for our purposes, since the search is case-insensitive. So far I don't see an example of your case ** here. What am I missing?

**Comment by jar398** [comment]
This seems to be fixed in OTT 3.0
